new old branch: Docs layout

ikbensiep opened the pull request:
Updates as discussed earlier today, includes mobile layout fix. 
This _should_ replace @mhum's earlier PR.

mhum committed:
Update to Jekyll v4

mhum committed:
Top nav bar and moved about to getting-started

mhum committed:
Move pages to new directory and update page styles

mhum committed:
Add middleware collection

mhum committed:
Dropdown middleware nav

mhum committed:
Begin documentation left nav

mhum committed:
Create getting started section

Commit:
Add jekyll to Gemfile (bundle would not run otherwise)

Commit:
Add Docs main menu item, manually set (getting started) docs order

Commit:
add curveball logo to header menu

Commit:
add doctype

Commit:
fix markup, fix writing

Commit:
Fix writing

Commit:
Update header menu layout

Commit:
move styles

Commit:
Update curveball logo (with tilted baseball)

Commit:
rename file (fix typo)

Commit:
Add permalink to include /docs, move some copy

Commit:
Add docs 'landing page'

Commit:
Add '/docs/' to permalink

Commit:
Add '/docs/' to permalink

Commit:
Add test command

Commit:
add 'stage 1' modules folder

Commit:
minor restructure of docs, add permalinks

Commit:
more restructuring / rewriting

Commit:
Update ALL the styles

Commit:
Update (Fav)icons

Commit:
Rewrite pages + includes

Commit:
remove unused files

Commit:
Add cc0 stock photos for 404 page

Commit:
update intermediary stylesheets

Commit:
Add some internal links

Commit:
Add classname to list

Commit:
remove shoutiness from badge

Commit:
fix styles for logo link

Commit:
Add badge styles, responsive sidebar system

Commit:
Updtae footer, homepage 'USPs'

Commit:
Update USPs layout, copy edits

Commit:
Updtae 404 page image

Commit:
Add sidebar toggle (for mobile view)

Commit:
Add link to logo

evert committed:
Docker container fixes

evert committed:
Remove unneeded lines

evert commented:
Small issue I noticed is that bullet points are gone:

![image](https://user-images.githubusercontent.com/178960/190842453-2fb5b605-8d2b-49e4-a617-9c30b63f4f8b.png)

evert commented:
A second (smallish) issue is that it's not super clear right now if an inline code block is clickable, it would be helpful if these get an underline:

![image](https://user-images.githubusercontent.com/178960/190842921-4373fb5b-56a2-4ed9-b9f3-5e709aa52133.png)

Example:

```
[`im code but also clickable`](https://evertpot.com/)
```

evert commented:
I actually noticed regular links also don't get underline in the main body. I'm really much in favor of always using underline for links that appear in text (and perhaps even always using a shade of blue, but i can leave it if you really dont like it). 

![image](https://user-images.githubusercontent.com/178960/190843026-ff8f7b24-b212-4336-8665-c248149f78de.png)

evert committed:
Docs from all middlewares

Copied mostly, but also tweaked a bit here and there.

ikbensiep committed:
Merge pull request #17 from curveball/middleware-docs

Docs from all middlewares

Commit:
Add max-width for img, code, add icon to page-nav links, make larger links

Commit:
Add underlines, bullets, small readability / whitespace improvements

Commit:
Add script to add target attribute to external links

Commit:
Add script to automatically generate page-nav links

Commit:
Remove unnecessary link, add h1 for SEO juice

Commit:
stylesheet updates propagated to assets

Commit:
Fix copy+markup

evert committed:
Merge branch 'docs_layout' into the-finest-docker-container

evert committed:
Merge remote-tracking branch 'origin/docs_layout' into the-finest-docker-container

Commit:
Add manual menu sorting

Commit:
Remove  from config

Commit:
Remove manual page-nav (replaced by js in documentation.html)

Commit:
fix little bug

Commit:
Change title

evert committed:
Merge pull request #18 from curveball/the-finest-docker-container

The finest docker container

evert committed:
Restructed docs a bit and wrote a bunch.

This can definitely use some proof-reading. It' quite hard to figure out
how much to explain.

evert committed:
Controller docs

Commit:
Add padding to page headers, mute page subtitle color

Commit:
update blockquote styles, add .note utility classes

Commit:
Fix markup, add highlighted note

Commit:
Fix main menu highligh in docs subpages

Commit:
Add lil' demo text options page

evert committed:
Merge pull request #19 from curveball/manual-sorting

Manual sorting

evert committed:
Merge pull request #20 from curveball/more-docs

Restructed docs a bit and wrote a bunch more

evert committed:
Merge branch 'main' into docs_layout
